vectordb_orm/base.py

```diff
@@ -0,0 +1,178 @@
+        collection = Collection(cls.collection_name(), schema)
+
+        # For all embeddings, create an associated index
+        for attribute_name, field_configuration in cls._type_configuration.items():
+            if isinstance(field_configuration, EmbeddingField):
+                index = {
+                    "index_type": field_configuration.index.index_type,
+                    "metric_type": field_configuration.metric_type.value,
+                    "params": field_configuration.index.get_parameters(),
+                }
+                collection.create_index("embedding", index)
+
+        return collection
+
+    @classmethod
+    def _field_schema_from_typehint(cls, name: str, type_hint: Any, field_customization: BaseField | None = None):
+        if issubclass(type_hint, np.ndarray):
+            if not isinstance(field_customization, EmbeddingField):
+                raise ValueError("Embedding typehints should be configured with vectordb.EmbeddingField")
+            return FieldSchema(name=name, dtype=DataType.FLOAT_VECTOR, dim=field_customization.dim)
+        elif issubclass(type_hint, str):
+            if not isinstance(field_customization, VarCharField):
+                raise ValueError("String typehints should be configured with vectordb.VarCharField")
+            return FieldSchema(name=name, dtype=DataType.VARCHAR, max_length=field_customization.max_length)
+        elif issubclass(type_hint, int):
+            # We currently only support ints as the primary key
+            is_primary = isinstance(field_customization, PrimaryKeyField)
+
+            # Mirror the python int sizing
+            return FieldSchema(name=name, dtype=DataType.INT64, is_primary=is_primary, auto_id=is_primary)
+        elif issubclass(type_hint, float):
+            # Mirror the python float sizing, equal to a double
+            return FieldSchema(name=name, dtype=DataType.DOUBLE)
+        else:
+            raise ValueError(f"{cls.__name__}: Typehint {name}:{type_hint} is not supported")
+
+    def insert(self, milvus_client: Milvus) -> int:
+        entities = self._dict_representation()
+        mutation_result = milvus_client.insert(collection_name=self.collection_name(), entities=entities)
+        self.id = mutation_result.primary_keys[0]
+        return self.id
+
+    def delete(self, milvus_client: Milvus) -> None:
+        if not self.id:
+            raise ValueError("Cannot delete object that hasn't been inserted into the database")
+
+        identifier_key = self._get_primary()
+        # Milvus only supports deleting entities with the `in` conditional; equality doesn't work
+        delete_expression = f"{identifier_key} in [{self.id}]"
+        milvus_client.delete(collection_name=self.collection_name(), expr=delete_expression)
+
+        self.id = None
+
+    def _dict_representation(self):
+        type_converters = {
+            np.ndarray: DataType.FLOAT_VECTOR,
```

Comment:
took me a bit of digging but it's probably worth highlighting that this is the one type conversion you have that Milvus does not support out of the box, and is a major improvement on it's ORM! #numpy is all you need